cranelift/bitset/src/compound.rs

```diff
@@ -0,0 +1,495 @@
+    ///```
+    pub fn capacity(&self) -> usize {
+        self.elems.capacity() * BITS_PER_WORD
+    }
+
+    /// Is this bitset empty?
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use cranelift_bitset::CompoundBitSet;
+    ///
+    /// let mut bitset = CompoundBitSet::new();
+    ///
+    /// assert!(bitset.is_empty());
+    ///
+    /// bitset.insert(1234);
+    ///
+    /// assert!(!bitset.is_empty());
+    /// ```
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+
+    /// Convert an element `i` into the `word` that can be used to index into
+    /// `self.elems` and the `bit` that can be tested in the
+    /// `ScalarBitSet<usize>` at `self.elems[word]`.
+    #[inline]
+    fn word_and_bit(i: usize) -> (usize, u8) {
+        let word = i / BITS_PER_WORD;
+        let bit = i % BITS_PER_WORD;
+        let bit = u8::try_from(bit).unwrap();
+        (word, bit)
+    }
+
+    /// The opposite of `word_and_bit`: convert the pair of an index into
+    /// `self.elems` and associated bit index into a set element.
+    #[inline]
+    fn elem(word: usize, bit: u8) -> usize {
+        let bit = usize::from(bit);
+        debug_assert!(bit < BITS_PER_WORD);
+        word * BITS_PER_WORD + bit
+    }
+
+    /// Is `i` contained in this bitset?
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use cranelift_bitset::CompoundBitSet;
+    ///
+    /// let mut bitset = CompoundBitSet::new();
+    ///
+    /// assert!(!bitset.contains(666));
+    ///
+    /// bitset.insert(666);
+    ///
+    /// assert!(bitset.contains(666));
+    /// ```
+    #[inline]
+    pub fn contains(&self, i: usize) -> bool {
+        let (word, bit) = Self::word_and_bit(i);
+        if word < self.elems.len() {
+            self.elems[word].contains(bit)
+        } else {
+            false
+        }
+    }
+
+    /// Reserve space in this bitset for the values `0..n`, growing the backing
+    /// storage if necessary.
+    ///
+    /// After calling `bitset.reserve(n)`, inserting any element `i` where `i <
+    /// n` is guaranteed to succeed without growing the bitset's backing
+    /// storage.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use cranelift_bitset::CompoundBitSet;
+    /// # let mut bitset = CompoundBitSet::new();
+    /// // We are going to do a series of inserts where `1000` will be the
+    /// // maximum value inserted. Make sure that our bitset has capacity for
+    /// // these elements once up front, to avoid growing the backing storage
+    /// // multiple times incrementally.
+    /// bitset.reserve(1001);
+    ///
+    /// for i in 0..=1000 {
+    ///     if i % 2 == 0 {
+    ///         // Inserting this value should not require growing the backing
+    ///         // storage.
+    ///         assert!(bitset.capacity() > i);
+    ///         bitset.insert(i);
+    ///     }
+    /// }
+    /// ```
+    #[inline]
+    pub fn reserve(&mut self, n: usize) {
```

Comment:
I might recommend renaming this method to `resize` or some variant thereof. The `Vec::reserve` method reserves space for extra capacity which isn't what this is doing so I'd be hesitant to create confusion between the two. Also `Vec::reserve` is mostly about preallocating uninitialized memory where this is specifically initializing the memory as well.